src/main.cpp

```diff
@@ -10,48 +11,62 @@
 
 WaggleNode node(RADIO_CE, RADIO_CSN);
 
-void reportSysStats() {
-    transmit_power_stats(node);
-}
-
 void reportTelemetry() {
+    Serial.println(F("Updating telemetry..."));
     collectData(node);
 }
 
-Ticker systemStatsTimer(reportSysStats, 10000);
+Ticker systemStatsTimer(updateSystemMonitor, 10000);
 Ticker telemetryTimer(reportTelemetry, 5000);
 
 void setup() {
+    wdt_enable(WDTO_8S);
+    signal_init();
+    signal_startup();
+    wdt_reset();
     // Setting up everything
     Serial.begin(115200);
+    Serial.print(F("RESET>\t"));
+    Serial.println(MCUSR, HEX);
+    MCUSR &= 0xf0;  // Clear the flag
     Serial.println(F("!>BOOT"));
     // Get DFU started
     if (is_dfu_available()) {
         Serial.println(F("?>D=DFU"));
         uint32_t flag = millis();
         while (millis() < flag + 1000) {
             if (Serial.available() && Serial.read() == 'D') {
-                while (1) dfu_loop();
+                while (1) {
+                    dfu_loop();
+                    wdt_reset();  // Hush doggy
+                }
             }
         }
     }
+
+    // Actually start a bunch of timeouts
+    wdt_reset();  // Hush doggy
     Serial.setTimeout(5000);
     analogReference(INTERNAL);
-    signal_init();
-    signal_startup();
     // Configure Net
     Serial.println(F(">>Starting node"));
     auto result = node.begin();
```

Comment:
@BoOoOoMsHaKaKa Check this. So basically just need to instantiate `node` like line 11 of this file, that's all of "activation" for the node.